extensions/frontend-tiles/src/GraphicsProvider/GraphicRepresentationProvider.ts

```diff
@@ -0,0 +1,263 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AccessToken, Logger} from "@itwin/core-bentley";
+import { loggerCategory} from "../LoggerCategory";
+
+/** The expected format of the Graphic Representation
+ * @beta
+ */
+/* eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents */
+export type GraphicRepresentationFormat = "IMDL" | "3DTILES" | string;
+
+/** Graphic representations are generated from Data Sources.
+ * The status of a Graphic Representation indicates the progress of that generation process.
+ * @beta
+ */
+export enum GraphicRepresentationStatus {
+  InProgress = "In progress",
+  Complete = "Complete",
+  NotStarted = "Not started",
+  Failed = "Failed",
+}
+
+/**
+ * Represents a data source for a graphic representation.
+ * A data source is usually higher-fidelity and contains more information, but may not be as well suited for visualization
+ * as a graphic representation.
+ * @beta
+ */
+export interface DataSource {
+  /** The iTwinId associated with the DataSource */
+  iTwinId: string;
+  /** The unique identifier of a DataSource.
+   * For example, a DataSource of type "IMODEL" has an iModelId which would be attributed to this value.
+   */
+  id: string;
+  /** The unique identifier for a specific version of a DataSource.
+   * For example, if a specific version of an iModel is desired, the iModel's changesetId would be attributed to this value.
+  */
+  changeId?: string;
+  /** The type of the data source. For example, a DataSource can be of type "IMODEL" or "RealityData" */
+  type: string;
+}
+
+/** Represents a visual representation of a data source, for example a 3d tileset.
+ * A data source can be an iModel, reality data, or other kind of graphical data.
+ * @see [[queryGraphicsDataSources]] for its construction as a representation of the data produced by a query of data sources.
```

Comment:
Just noticed some of the documentation and test descriptions still have the old naming which I would suggest fixing. There are 2 instances in this file, and all the others are all in GraphicRepresentationProvider.test.ts. I just searched for "GraphicsDataSource" and "graphics data source" to find them

Reply:
Good catch, they should all be updated now.